server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -156,54 +167,164 @@ private void finishLeaf() {
         }
     }
 
+    /** Return true if the provided field has multiple values per document in the leaf **/
+    private boolean isMultiValued(LeafReaderContext context, SortField sortField) throws IOException {
+        SortField.Type type = IndexSortConfig.getSortFieldType(sortField);
+        if (type == SortField.Type.STRING) {
+            final SortedSetDocValues values = context.reader().getSortedSetDocValues(sortField.getField());
+            if (values != null && DocValues.unwrapSingleton(values) != null) {
+                return false;
+            }
+        } else {
```

Comment:
can you check for the numeric types explicitly? I worry that we might end up in that method with a SortField.Type.CUSTOM or DOC?

Reply:
Sure, I pushed https://github.com/elastic/elasticsearch/pull/48399/commits/03141a547a3011fa4fa49c60a58e59d85ec520e0